### ISSUE TITLE
\pkg@error{x}{x} prints '\MessageBreak x'

### ISSUE THREAD

**Issue by godbyk** [wspr/hardwrap#8]
Here's the test case:

```
\GeneratePackageLogMacros{test}
\test@error{x}{x}
```

Output is:

```
! Package test Error: \MessageBreak x.
```

Similarly for the class version.

(info and warning appear okay, though.

**Comment by godbyk** [comment]
Fixed in 055d068ea3ad301e38a3f0860471b48c25511ed6.

If those `\MessageBreak`s were there for a good reason, let me know.

**Comment by godbyk** [comment]
Closing.

**Comment by wspr** [comment]
They were there for a reason, but they were obviously not being printed correctly! The problem is if `-line-file-error` is enabled then you get error messages that look like

```
./hardwrap.dtx:211: Package hardwrap Error: [...]
```

where the prefix before the line number is of arbitrary length. So this needs to be broken or else our calculations are messed up.

I'll try and work out what I was trying to do originally and why the messagebreak wasn't going through properly.

**Comment by godbyk** [comment]
Ah, I see.  Lemme guess -- we can't detect if we're running in file:line:error mode, can we?  If we can, we should suppress wrapping the error messages so they can be parsed by the IDE/editor.

**Comment by godbyk** [comment]
We could grab file_line_error_style from texmf.cnf, but I that won't help us if they override the default on the tex command line.

**Comment by wspr** [comment]
code dump

sorry! I was working on this on-and-off today
and the whole thing turned into a mess

I hope this doesn't get in your way too much

closed by 6c0119798d9e755caa1244e3221ea89e0e42cae8
closed by 6c0119798d9e755caa1244e3221ea89e0e42cae8
